app/hotels/src/navigation/__tests__/AllHotelsNavigationScreen.test.js

```diff
@@ -1,37 +1,22 @@
 // @flow
 
-import * as React from 'react';
 import { Platform, BackHandler } from 'react-native';
-import { shallow } from 'enzyme';
-
-import { AllHotelsNavigationScreen } from '../AllHotelsNavigationScreen';
-
-const props = {
-  navigation: {},
-  currency: 'EUR',
-  coordinates: null,
-  onSearchChange: jest.fn(),
-  onFilterChange: jest.fn(),
-  onLocationChange: jest.fn(),
-  onCityIdChange: jest.fn(),
-};
-
-const renderNavigationScreen = (onBackClicked: Function) => {
-  const renderProps = {
-    ...props,
-    onBackClicked,
-  };
-  // $FlowIssue: https://github.com/facebook/flow/issues/2405
-  return shallow(<AllHotelsNavigationScreen {...renderProps} />);
-};
+
+import {
+  registerBackButtonListener,
+  resetBackButtonListener,
+} from '../AllHotelsNavigationScreen';
 
 describe('AllHotelsNavigationScreen', () => {
+  afterEach(() => {
+    resetBackButtonListener();
+  });
   it('registers an event listener if platform is android', () => {
     const onBackClicked = jest.fn();
     Platform.OS = 'android';
```

Comment:
I didn't notice this before but it's necessary to reset `OS` back when we overwite global variable.